internal/kubernetes/reflectionController.go

```diff
@@ -112,6 +112,18 @@ func (p *KubernetesProvider) controlLoop() {
 
 func (p *KubernetesProvider) cleanupNamespace(ns string) error {
 
+	pods, err := p.foreignClient.Client().CoreV1().Pods(ns).List(metav1.ListOptions{})
```

Comment:
You can use provider's foreign caches instead of performing this API server request